notifications_api_common/viewsets.py

```diff
@@ -171,37 +171,17 @@ def notify(
         # We've performed all the work that can raise uncaught exceptions that we can
         # still put inside an atomic transaction block. Next, we schedule the actual
         # sending block, which allows failures that are logged. Any unexpected errors
-        # here will still cause the transaction to be comitted (in the default
+        # here will still cause the transaction to be committed (in the default
         # behaviour), but the exception will be visible in the error monitoring (such
         # as Sentry).
         #
-        # The _send function is passed down to the scheduler, which is by default to
-        # execute it on transaction commit.
+        # The 'send_notification' task is passed down to the task queue on transaction commit
 
         def _send():
-            try:
-                client.create("notificaties", message)
-            # any unexpected errors should show up in error-monitoring, so we only
-            # catch ClientError exceptions
-            except ClientError:
-                logger.warning(
-                    "Could not deliver message to %s",
-                    client.base_url,
-                    exc_info=True,
-                    extra={
-                        "notification_msg": message,
-                        "status_code": status_code,
-                    },
-                )
-
-        self.schedule_notification(_send)
-
-    @staticmethod
-    def schedule_notification(send_function: callable):
-        """
-        Ensure that a notification is scheduled to be sent.
-        """
-        transaction.on_commit(send_function)
+            send_notification.delay(message)
+
+        # transaction.on_commit(_send)
+        send_notification.delay(message)
```

Comment:
this looks wrong - now the notification is still scheduled in celery even if the DB transaction will be rolled back.

Shouldn't this just be `transaction.on_commit(lambda: send_notification.delay(message))`?

Reply:
oops, I changed it back and forth when I was testing it, and now it's not correct. it should be `transaction.on_commit(_send)`
I'll change it according to your suggestion